packages/graphql-language-service-server/src/MessageProcessor.ts

```diff
@@ -288,6 +288,39 @@ export class MessageProcessor {
     );
   }
 
+  async _getDiagnosticsForAllFileProjects(
```

Comment:
This method is extracted from the code below to avoid repetition, and it's adapted to run for every project that matches the file.